src/parsing_visualizer.cpp

```diff
@@ -0,0 +1,172 @@
+#include "include/parsing_visualizer.hpp"
+#include "include/parser.hpp"
+#include <vector>
+#include "gvpp.hpp"
+#include <iostream>
+#include <fstream>
+
+std::string stringify_op(const expression& expr, const char* sep, const char* delim, bool stmt, unsigned first, unsigned limit) {
```

Comment:
This function is really messy. Way too many parameters with unclear names and wrong types (prefer `const std::string &` over `const char *`)

Reply:
I have some doubts if it's worth to replace this ```const char *``` with ```std::string &``` since they arе one to three characters only and are not treated like strings but like chars with two spaces on both left and right sides

Reply:
There are no casts, because there is a `std::string` constructor from `const char *`.